cpp/ql/src/Diagnostics/FailedCompilations.ql

```diff
@@ -0,0 +1,22 @@
+/**
+ * @name Failed compiler invocations.
+ * @description Gives the command-line of compilations for which extraction did not run to completion.
+ * @kind diagnostic
+ * @id cpp/diagnostics/failed-compilations
```

Comment:
I'm not happy about that, because `SuccessfulExtractions` and `FailedExtractions` should be disjoint sets, so that the union of these is the set of all files in the repository.
Maybe your query can be `AbortedExtractions`? That would add another word besides `Unrecoverabe`, so that's a bit meh.
Or `UnrecoverablyFailedExtraction`? But that, I think, is not clear enough.

Reply:
I suppose we're implicitly using "extraction" to mean "of a single file", then, whereas this query is for the whole command line.

How about `FailedInvocations` or `FailedExtractorInvocations` then? Or maybe `Incomplete` rather than `Failed`?

Reply:
`FailedInvocations` it is - I like it most, because it doesn't have `extractor/extraction` in its name, so the query is less likely to be confused with the others.
`Incomplete` also suggests to me that we extracted at least something, which (at least) in the case of an EDG assertion or hitting the error limit might not be true.